src/passes/Precompute.cpp

```diff
@@ -41,6 +41,12 @@ namespace wasm {
 
 static const Name NOTPRECOMPUTABLE_FLOW("Binaryen|notprecomputable");
 
+// Limit evaluation depth for 2 reasons: first, it is highly unlikely
+// that we can do anything useful to precompute a hugely nested expression
+// (we should succed at smaller parts of it first). Second, a low limit is
+// helpful to avoid platform differences in native stack sizes.
+enum { MAX_DEPTH = 50 };
```

Comment:
Why use an enum here?

Reply:
It could be a `static const` or a `#define` I guess. I don't have a preference, what do you think?

Reply:
I would default to using a `static const`. That way the underlying type is clear and there are no extra type shenanigans happening.